src/main/java/com/google/sps/servlets/BuddyServlet.java

```diff
@@ -0,0 +1,56 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.Constants;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/buddy")
+public class BuddyServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    String currentUserId = userService.getCurrentUser().getUserId();
+    String otherUserId = request.getParameter(Constants.BUDDY_ADD_PARAM);
+
+    Entity currentUserEntity = getUserEntity(currentUserId);
+    Entity otherUserEntity = getUserEntity(otherUserId);
+
+    List<String> currentBuddiesList = (List<String>) currentUserEntity.getProperty(Constants.USER_BUDDIES_PARAM);
+    List<String> otherBuddiesList = (List<String>) otherUserEntity.getProperty(Constants.USER_BUDDIES_PARAM);
+
+    // If the users are already buddies, remove buddies.
+    if (currentBuddiesList.contains(otherUserId)) {
```

Comment:
I'm not sure I follow this logic. Why remove the users when they are already buddies?

Reply:
I'm planning on calling this function whenever the user clicks the add or remove buddy button. If the users are already buddies, that means that this function is being called with the intention of removing the buddy relationship. If the users are not yet buddies, that means that this function is being called with the intention of adding the buddy relationship. Is there a different way I should go about things to make it more clear? (I did something similar with the Interest servlet, too)

Reply:
That's a good question. Fine for now. 

If you plan to refactor this class, then I would consider making the contract more explicit. Instead using this as a toggle, have two explicit calls for adding and removing a buddy. Even though some of the scaffolding code might be duplicated, in the long run being explicit about state makes the code more readable, easier to reason about, and easier to debug.